src/Rox/Core/Roxx/StringTokenizer.php

```diff
@@ -149,7 +149,7 @@ private static function isSurrogate($c)
      */
     private static function charCount($codePoint)
     {
-        return $codePoint >= self::MIN_SUPPLEMENTARY_CODE_POINT ? 2 : 1;
+        return ord($codePoint) >= self::MIN_SUPPLEMENTARY_CODE_POINT ? 2 : 1;
```

Comment:
Although undocumented, it seems `string X number` and `number X string` comparison (not just equality) changed in a poorly defined way. Kinda surprised, but I guess Rox has a full blown parser and it has a couple of questionable usages of this. According to the comments, the affected code is largely supposed to just be an "optimization". I'm about 90% confident that this change is all that needs to be made. The tests are also pretty well written.

PHP 7.4

```php
echo(ord("{"));   // => 123
echo((int) "{");  // => 0

echo(-1  < "{" ? 'yes' : 'no'); // => 'yes'
echo( 0  < "{" ? 'yes' : 'no'); // => 'no'
echo(999 < "{" ? 'yes' : 'no'); // => 'no'
```

PHP 8.0

```php
echo(ord("{"));   // => 123
echo((int) "{");  // => 0

echo(-1  < "{" ? 'yes' : 'no'); // => 'yes'
echo( 0  < "{" ? 'yes' : 'no'); // => 'yes'
echo(999 < "{" ? 'yes' : 'no'); // => 'yes'
```

Both of these results are silly, but so are dynamic languages for ever allowing this to be a thing. I bet this got introduced in https://wiki.php.net/rfc/string_to_number_comparison. And, unrelated, but I love how clean that Request For Change is and that they make decisions with a public, "strong" majority voting system is pretty cool.